src/app/shared/settings/scroll.component.ts

```diff
@@ -0,0 +1,144 @@
+					this.setBtnScroll();
+					this.modalBtn.setAttribute('data-value', 'Boutons ascenseurs au survol');
+					break;
+				}
+				default: {
+					this.resetScroll();
+					this.modalBtn.setAttribute('data-value', 'default');
+				}
+			}
+		});
+	}
+
+	disconnectedCallback(): void {
+		super.disconnectedCallback();
+		this.settingBtn?.removeEventListener('changeSettingEvent', () => { });
+		this.btnScrollUp?.removeEventListener('click', () => { });
+		this.btnScrollUp?.removeEventListener('mouseover', () => { });
+		this.btnScrollDown?.removeEventListener('click', () => { });
+		this.btnScrollDown?.removeEventListener('mouseover', () => { });
+	}
+
+	setScrollClass = (): void => {
+		let classScroll = `
+			.cplus-big-scroll::-webkit-scrollbar, .cplus-big-scroll *::-webkit-scrollbar {
+					width: 2rem;
+			}
+			.cplus-big-scroll::-webkit-scrollbar-thumb, .cplus-big-scroll *::-webkit-scrollbar-thumb {
+				background-color: lightgrey;
+				border-radius: 1.75rem
+				width: 2rem;
+				cursor: pointer;
+			}
+			.cplus-big-scroll::-webkit-scrollbar-thumb:hover, .cplus-big-scroll *::-webkit-scrollbar-thumb:hover {
+				background-color: grey;
+			}
+
+			#cplus-container-scroll-buttons {
+				display: flex;
+				gap: 1rem;
+				position: fixed;
+				bottom: 1rem;
+				right: 1rem;
+				z-index: 2147483647;
+			}
+		`;
+
+		let head: HTMLHeadElement = document.head || document.getElementsByTagName('head')[0];
+		let stylesScroll: HTMLStyleElement = document.createElement('style');
+		stylesScroll.setAttribute('id', 'cplus-scroll');
+		stylesScroll.innerHTML = classScroll;
+		head.appendChild(stylesScroll);
+	}
+
+	setBigScroll = (): void => {
+		this.bodyElt.classList.add('cplus-big-scroll');
+	}
+
+	setBtnScroll = (): void => {
+		const bodyElt: HTMLBodyElement = document.getElementsByTagName('body')[0];
+		const container = document.createElement('div');
+		container.setAttribute('id', 'cplus-container-scroll-buttons')
+		let btnArray: any[] = [];
+
+		let btnUp = `<button id="cplus-scroll-up" class="btn btn-primary">Monter</button>`;
+		let btnDown = `<button id="cplus-scroll-down" class="btn btn-primary">Descendre</button>`;
+
+		btnArray.push(btnUp, btnDown);
+		container.innerHTML = btnArray.join('');
+		bodyElt.appendChild(container);
+
+		this.btnScrollUp = document.querySelector('#cplus-scroll-up');
+		this.btnScrollDown = document.querySelector('#cplus-scroll-down');
+
+		if (this.btnState === 'click') {
+			this.btnScrollUp.addEventListener('click', (event: any) => {
+				window.scrollBy(0, -this.scrollSteps);
+			});
+			this.btnScrollDown.addEventListener('click', (event: any) => {
+				window.scrollBy(0, this.scrollSteps);
+			});
+		} else if (this.btnState === 'mouseover') {
+			this.btnScrollUp.addEventListener('mouseover', (event: any) => {
+				window.scrollBy(0, -this.scrollSteps);
+			});
+			this.btnScrollDown.addEventListener('mouseover', (event: any) => {
+				window.scrollBy(0, this.scrollSteps);
+			});
+		}
```

Comment:
On ne pourrait pas simplement passer `this.btnState` en argument du `addEventListener`, plutôt que de répéter le bloc de code ? À tester…